# Type: fix
# Subtype: I3
# English Summary: Modified if conditional statement
# Chinese Summary: 修改if条件语句
# Entry Number: 22

# === Git Diff Content ===

diff --git a/platform/core-impl/src/com/intellij/psi/impl/source/PsiFileImpl.java b/platform/core-impl/src/com/intellij/psi/impl/source/PsiFileImpl.java
index 0315ce5..b7e395d 100644
--- a/platform/core-impl/src/com/intellij/psi/impl/source/PsiFileImpl.java
+++ b/platform/core-impl/src/com/intellij/psi/impl/source/PsiFileImpl.java
@@ -253,7 +253,7 @@ public abstract class PsiFileImpl extends ElementBase implements PsiFileEx, PsiF
     synchronized (myStubLock) {
       StubTree fileStub = derefStub();
       StubElement childStub = child.getStub();
-      if (fileStub == null || childStub != null && fileStub.getRoot() != childStub.getParentStub()) {
+      if (childStub != null && (fileStub == null || fileStub.getRoot() != childStub.getParentStub())) {
         return false;
       }
     }
